test/postgres-conflict.test.ts

```diff
@@ -0,0 +1,186 @@
+/* eslint-disable @typescript-eslint/unbound-method */
+import { Prisma, PrismaClient } from '@prisma/client';
+import * as Budget from 'document-model-libs/budget-statement';
+import { utils } from 'document-model-libs/document-drive';
+import { DocumentHeader, Operation, State } from 'document-model/document';
+import { setTimeout } from 'node:timers/promises';
+import { beforeAll, describe, it, vi } from 'vitest';
+import { generateUUID } from '../src';
+import { PrismaStorage } from '../src/storage/prisma';
+import { DocumentStorage } from '../src/storage/types';
+import { buildOperationAndDocument } from './utils';
+
+describe('Storage with cache proxy', () => {
+    const client = new PrismaClient<
+        Prisma.PrismaClientOptions,
+        Prisma.LogLevel
+    >({ log: ['error'] });
+    const storage = new PrismaStorage(client);
+    const DRIVE_ID = 'postgres-conflict-test';
+
+    beforeAll(async () => {
+        const drives = await storage.getDrives();
+        if (drives.includes(DRIVE_ID)) {
+            await storage.deleteDrive(DRIVE_ID);
+        }
+        const newDrive = utils.createDocument();
+        await storage.createDrive(DRIVE_ID, newDrive);
+    });
+
+    it('should not have conflict on operations for two separate documents', async ({
+        expect
+    }) => {
+        await storage.createDocument(
+            DRIVE_ID,
+            'doc1',
+            Budget.utils.createDocument(),
+            [{ syncId: '1', scope: 'global', branch: 'main' }]
+        );
+        await storage.createDocument(
+            DRIVE_ID,
+            'doc2',
+            Budget.utils.createDocument(),
+            [{ syncId: '2', scope: 'global', branch: 'main' }]
+        );
+
+        async function callback(documentStorage: DocumentStorage): Promise<{
+            operations: Operation[];
+            header: DocumentHeader;
+            newState?: State<any, any> | undefined;
+        }> {
+            const { document, operation } = buildOperationAndDocument(
+                Budget.reducer,
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+                Budget.utils.createDocument(documentStorage as any),
+                Budget.actions.addAccount({ address: '0x123' })
+            );
+            await setTimeout(100);
+            operation.id = generateUUID();
+            operation.resultingState = document.state[operation.scope];
+            return Promise.resolve({
+                header: document,
+                operations: [operation],
+                newState: document.state
+            });
+        }
+
+        const callbackSpy = vi.fn(callback);
+
+        let prismaError: Prisma.LogEvent | undefined;
+        client.$on('error', error => {
+            prismaError = error;
+        });
+
+        const results = await Promise.all([
```

Comment:
This was not working before, concurrent operations to different documents were causing the latest op to retry.